src/common/ad_tx.c

```diff
@@ -8,29 +8,60 @@
 #include <daos_srv/ad_mem.h>
 #include "ad_mem.h"
 
+static char *
+act_opc2str(int act)
+{
+	switch (act) {
+	default:
+		D_ASSERTF(0, "unknown opcode=%d\n", act);
+		return "unknown";
+	case UMEM_ACT_NOOP:
```

Comment:
this is where I really like to use FOREACH macros so you have an array of strings where UMEM_ACT_NOOP points to "NOOP", etc.  It can make a big difference in overhead of TRACE